include/turnstile.h

```diff
@@ -10,7 +10,8 @@ void turnstile_init(turnstile_t *);
 
 /* This puts currently running thread on the turnstile,
  * puts it in waiting state and yields. While waiting on turnstile
- * thread cannot be run. This function is done under critical section. */
+ * thread cannot be run. This function has to be called with critical section,
```

Comment:
Sorry to disappoint you about not asking ;).

I like this solution more. I just want to mention that FreeBSD counterpart requires we enter the `turnstile wait` with spinlock on turnstile, which in our case is equivalent to entering this function in with critical section on.